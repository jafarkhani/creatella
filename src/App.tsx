
import ProductGrid from './components/ProductGrid/ProductGrid';

const App:React.FC = () => {
  return (
    <div className="container-fluid mt-2 mb-2">
      <ProductGrid />
    </div>
  );
}

export default App;
