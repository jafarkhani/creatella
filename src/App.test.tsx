
import React from 'react';
import { cleanup } from '@testing-library/react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import App from './App';

describe("App component tests", ()=>{

  let wrapper:any;

  beforeEach(() => {
    wrapper = shallow(<App />)
  })

  afterEach(cleanup); 

  it("renders without crashing", () => {
      expect(wrapper.length).toBe(1)
  });

  it("renders correctly", () => {
    expect(wrapper.type()).toEqual('div');
  });

  it('snapshop App', () => {
      const tree = renderer.create(<App />).toJSON();
      expect(tree).toMatchSnapshot();         
  });
  
});