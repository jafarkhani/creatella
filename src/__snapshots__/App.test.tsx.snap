// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component tests snapshop App 1`] = `
<div
  className="container-fluid mt-2 mb-2"
>
  <div>
    <table
      className="table table-striped table-bordered"
    >
      <thead>
        <tr
          data-testid="headers"
        >
          <th>
            <button
              className="sortable-header"
              onClick={[Function]}
            >
              ID
            </button>
          </th>
          <th>
            Face
          </th>
          <th>
            <button
              className="sortable-header"
              onClick={[Function]}
            >
              Price
            </button>
          </th>
          <th>
            <button
              className="sortable-header"
              onClick={[Function]}
            >
              Size
            </button>
          </th>
          <th>
            Added Date
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            align="center"
            colSpan={5}
          >
            No Record founds
          </td>
        </tr>
      </tbody>
      <tfoot />
    </table>
  </div>
</div>
`;
